flame_utils/io/lattice.py

```diff
@@ -77,8 +78,6 @@ def generate_latfile(machine, state=None, latfile=None, out=None):
         if type(state) == type(m.allocState({})):
```

Comment:
How about change this line to be `if isinstance(state, (MachineStates, flame._internal.State)):` to include the case of `state` is a `MachineStates` object?

Then at the top of this file, should add import lines:
```python
import flame
from flame_utils.core import MachineStates
```